### DISCUSSION TITLE
Geting battery percentage

### DISCUSSION THREAD

**Discussion by kozolejure** [mikey0000/PyMammotion#14] [General]
I've checked the API for Bluetooth connection, but I'm wondering if it's possible to obtain battery status information, such as the level of charge?

**Comment by mikey0000** [comment]
Yeah totally is. Not really sure what you mean by checked the api, the pyluba library contains all protobuf messages, so if you decompile the apk and look at esputil you'll see all the commands, there are a lot of commands already in pyluba, the main protobuf message that comes back from Luba I'm pretty sure has battery info, Luba will start sending them as soon as you connect over ble.

Part of pyluba that needs further work is converting those messages from protobuf back into models, which is partly there.

**Reply by mikey0000** [reply]
turns out I made a small mistake in my statement, Luba sends the data only when its working.

**Comment by mikey0000** [comment]
Finally worked out which command gets back the most useful data and its get_report_cfg can find example in test_instance.py

**Reply by kozolejure** [reply]
Thanks man, this week i will have more time so i will try to implement features like battery level reporting and etcetera. I will try to use a device like esp32 or something similar. Do you think this is possible?

**Reply by mikey0000** [reply]
I mean the Luba has an esp32 in it so yeah. But trying to port the python code over to an esp32 is probably a tall order. A raspberry pi with esp32 Bluetooth proxies is the direction I'd head down. Home assistant already provides this which is why I'm working on the Home Assistant integration.
